crates/pkgcraft/src/eapi.rs

```diff
@@ -24,32 +25,79 @@ use crate::shell::hooks::HookBuilder;
 use crate::shell::operations::{Operation, OperationKind};
 use crate::shell::phase::Phase;
 
-peg::parser!(grammar parse() for str {
-    // EAPIs must not begin with a hyphen, dot, or plus sign.
-    pub(super) rule eapi() -> &'input str
-        = s:$(quiet!{
-            ['a'..='z' | 'A'..='Z' | '0'..='9' | '_']
-            ['a'..='z' | 'A'..='Z' | '0'..='9' | '+' | '_' | '.' | '-']*
-        } / expected!("EAPI"))
-        { s }
-
-    rule single_quotes<T>(expr: rule<T>) -> T
-        = "\'" v:expr() "\'" { v }
+mod parser {
+    use winnow::combinator::{alt, delimited, trace};
+    use winnow::prelude::*;
+    use winnow::token::{one_of, take_while};
+
+    const APOSTROPHE: char = '\'';
+    const QUOTATION_MARK: char = '"';
+
+    /// Parse an eapi, specified in PMS 3.1.9.
```

Comment:
I'm not sure how stable spec section numbers are across revisions.

Anyway, it would probably be more helpful to those that are unfamiliar with it to give a short description rather than a reference since it should be simple enough to explain.